MaterialLib/PorousMedium/UnsaturatedProperty/CapillaryPressure/VanGenuchtenCapillaryPressureSaturation.cpp

```diff
@@ -77,6 +76,11 @@ double VanGenuchtenCapillaryPressureSaturation::getdPcdS(
 
         return -getdPcdSvGBar(1 - _saturation_r);
     }
+    if (saturation < _saturation_r)
+        return 0;
+    if (saturation > _saturation_max)
+        return 0;
```

Comment:
Why not `_saturation_max`?

Reply:
Same cut-off in function => derivative = 0